From: Adam Jackson <ajax@redhat.com>
Date: Mon, 10 Mar 2008 13:08:21 +0000 (-0400)
Subject: C sucks: define XEventClass in terms of unsigned int, not CARD32.
X-Git-Url: http://gitweb.freedesktop.org/?p=xorg/proto/inputproto.git;a=commitdiff;h=b5cbe2d93f6c0129b8f29da97778f6d1b15c38f9

C sucks: define XEventClass in terms of unsigned int, not CARD32.

Apparently pulling in Xmd.h here breaks qt, since they both define an
INT32 type (and incompatible ones even, since Xmd's is unsigned long on
ILP32 because whoever wrote Xmd.h is a C novice).
--- a/XI.h
+++ b/XI.h
@@ -1,5 +1,3 @@
-/* $Xorg: XI.h,v 1.4 2001/02/09 02:03:23 xorgcvs Exp $ */
-
 /************************************************************
 
 Copyright 1989, 1998  The Open Group
@@ -45,17 +43,12 @@ ARISING OUT OF OR IN CONNECTION WITH THE
 SOFTWARE.
 
 ********************************************************/
-/* $XFree86: xc/include/extensions/XI.h,v 1.4 2001/01/17 17:53:16 dawes Exp $ */
 
 /* Definitions used by the server, library and client */
 
 #ifndef _XI_H_
 #define _XI_H_
 
-#include <X11/Xmd.h> /* CARD32 */
-
-#define sz_xGetExtensionVersionReq		8
-#define sz_xGetExtensionVersionReply		32
 #define sz_xListInputDevicesReq			4
 #define sz_xListInputDevicesReply		32
 #define sz_xOpenDeviceReq			8
@@ -263,12 +256,16 @@ SOFTWARE.
 #define XI_DeviceBusy	3
 #define XI_BadClass	4
 
-/* Make XEventClass be a CARD32 for 64 bit servers.  Don't affect client
+/*
+ * Make XEventClass be a CARD32 for 64 bit servers.  Don't affect client
  * definition of XEventClass since that would be a library interface change.
  * See the top of X.h for more _XSERVER64 magic.
+ *
+ * But, don't actually use the CARD32 type.  We can't get it defined here
+ * without polluting the namespace.
  */
 #ifdef _XSERVER64
-typedef	CARD32		XEventClass;
+typedef unsigned int	XEventClass;
 #else
 typedef	unsigned long	XEventClass;
 #endif
From: Adam Jackson <ajax@redhat.com>
Date: Mon, 10 Mar 2008 13:31:51 +0000 (-0400)
Subject: Typo fix.
X-Git-Url: http://gitweb.freedesktop.org/?p=xorg/proto/inputproto.git;a=commitdiff;h=852568991b251e9366da167f1b746a0a1db6adf0

Typo fix.
--- a/XI.h
+++ b/XI.h
@@ -49,6 +49,8 @@ SOFTWARE.
 #ifndef _XI_H_
 #define _XI_H_
 
+#define sz_xGetExtensionVersionReq             8
+#define sz_xGetExtensionVersionReply           32
 #define sz_xListInputDevicesReq			4
 #define sz_xListInputDevicesReply		32
 #define sz_xOpenDeviceReq			8
